### DISCUSSION TITLE
fixed_date - photo_date

### DISCUSSION THREAD

**Discussion by bimbamborius** [RhetTbull/osxphotos#1701] [Q&A] [answered]
👋📷🤓
I want to calculate the time difference between the date created of photos of my daughter and her date of birth.
How would i achieve this? Do i need a {function} from the template substitutions for this? 
I'm new to osxphotos and don't have experience with python. 
Any help is appreciated 🙏
And thanks for your amazing work 💫

**Comment by RhetTbull** [comment, marked as answer]
Hi @bimbamborius , glad you've found OSXPhotos useful. There's no way to do this out of the box but I've created a [simple template function](https://github.com/RhetTbull/osxphotos/blob/main/examples/date_delta.py) that might do what you want. There are three functions defined in [date_delta.py](https://github.com/RhetTbull/osxphotos/blob/main/examples/date_delta.py): `years_since`, `months_since`, and `days_since`. To use them in any osxphotos command that accepts templates, you'd call like this:

`"{function:https://raw.githubusercontent.com/RhetTbull/osxphotos/refs/heads/main/examples/date_delta.py::months_since(2021-10-01)}"`

Alternatively, you could download [date_delta.py](https://raw.githubusercontent.com/RhetTbull/osxphotos/refs/heads/main/examples/date_delta.py) to your Mac and call it like this:

`"{function:date_delta.py::months_since(2021-10-01)}"`
`"{function:date_delta.py::days_since(2021-10-01)}"`
 `"{function:date_delta.py::years_since(2021-10-01)}"`

The date argument is passed in `YYYY-MM-DD` format.

For example, this query command prints the photo name, the photo date, and the date delta in months. Note that the date passed is assumed to be local timezone and the photo date will use the timezone of the photo so this could mean that the delta is different than expected when accounting for timezones. 

`osxphotos query --quiet --print "{photo.original_filename} {photo.date} {function:https://raw.githubusercontent.com/RhetTbull/osxphotos/refs/heads/main/examples/date_delta.py::months_since(2021-10-01)}"`
